bindings/rust/extended/s2n-tls/src/connection.rs

```diff
@@ -1000,6 +1000,34 @@ impl Connection {
         }
     }
 
+    pub fn kem_group_name(&self) -> Option<&str> {
```

Comment:
Since these are the bindings and we're not necessarily locked into the same choices we made for the C: do we want to just change the definition of kem_name or selected_curve instead? I remember you had a document about that.

Reply:
Although making those three methods behave differently in the bindings vs the C might make an already confusing situation worse. Maybe a very bad idea.

Reply:
Agreed, the C and Rust API's should have matching behavior. 

My recommended fix for this is to add a `s2n_connection_get_group_name` API and deprecate everything else, because that's what customers likely **actually** want. https://github.com/aws/s2n-tls/issues/4123 

They probably don't want to call one API for ECDHE and another for PQ, and call both if Hybrid is used. They likely want s2n to handle all of that for them.